clang/test/SemaCXX/type-traits.cpp

```diff
@@ -1741,8 +1741,10 @@ void is_layout_compatible(int n)
   static_assert(!__is_layout_compatible(unsigned char, signed char));
   static_assert(__is_layout_compatible(int[], int[]));
   static_assert(__is_layout_compatible(int[2], int[2]));
-  static_assert(!__is_layout_compatible(int[n], int[2])); // FIXME: VLAs should be rejected
-  static_assert(!__is_layout_compatible(int[n], int[n])); // FIXME: VLAs should be rejected
+  static_assert(!__is_layout_compatible(int[n], int[2]));
+  // expected-error@-1 {{variable length arrays are not supported for '__is_layout_compatible'}}
+  static_assert(!__is_layout_compatible(int[n], int[n]));
```

Comment:
I think where we ended up is:

1) Reject incomplete types (still needs to be done)
2) Reject VLAs (done, looks good)
3) FAMs are accepted (no changes needed)

So I think you should implement #1 and add tests for it, then add tests for #3, and I think that finishes this.

Reply:
I recognize that discussion has spiraled out of VLAs into incomplete types and flexible array members, but can I keep the original scope of this PR, focusing on just VLAs?

Reply:
I'm fine with splitting the work across multiple patches so long as we get to the desired end state (or have issues filed to track what that end state should be).

Reply:
Yes, other items will be addressed in subsequent PRs.

Reply:
I think focusing on VLAs here is fine, that was the original scope and I don't see an urgent need to expand it here.